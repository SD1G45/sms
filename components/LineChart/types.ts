export interface LineChartProps extends React.HTMLAttributes<HTMLInputElement> {
  title: string;
  data: {[key: string]: number | string}[];
  height:  number;
  flexure: number;
}