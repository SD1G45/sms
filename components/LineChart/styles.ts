import styled from "styled-components";
import { Path, TooltipWrapper, HintPoint, TooltipLabel, TooltipValue, TooltipXAxisValue } from "styled-chart";

export const StyledPath = styled(Path)
    <{strokeColor: string,
      strokeWidth?: number,
      fillColor?: string,
    }>`
    && {
      ${({ strokeColor, strokeWidth, fillColor }) =>`
        stroke: ${strokeColor} !important;
        stroke-width: ${strokeWidth ? strokeWidth : 2}px;
        fill: ${fillColor ? fillColor : 'transparent'};
      `}
    }
  `

  export const StyledHint = styled(HintPoint)<{backgroundColor?: string, textColor?: string}>`
    && {
      ${({ backgroundColor }) => `
        background-color: ${backgroundColor};
      `}
    }
  `

  export const StyledTooltip = styled(TooltipWrapper)
  <{backgroundColor?: string,
    textColor?: string,
  }>`
    && {
      ${({ backgroundColor, textColor }) => `
        ${backgroundColor ? `background: ${backgroundColor}` : ``};
        ${textColor ? `color: ${textColor}` : ``};
      `}
      ${TooltipLabel} {
        font-style: italic;
      }
      ${TooltipValue} {
        font-style: italic;
      }
      ${TooltipXAxisValue} {
        opacity: 0.8;
      }
      ${StyledHint} {
        && {
          padding-left: 4px;
          padding-right: 4px;
          border-radius: 20px;
          height: 3px;
        }
      }
    }
  `

  export const StyledTitle = styled.text`
    text-align: left;
    margin-bottom: -1em;
  `;

  export const ChartWrapper = styled.div`
    display: flex;
    flex-direction: column;
    width: 100%;
    gap: 10%;
  `;