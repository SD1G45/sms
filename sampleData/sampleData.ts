function couponData() {
  return [
    {
      date: "12:00 AM",
      "coupons": 0
    },
    {
      date: "1:00 AM",
      "coupons": 0
    },
    {
      date: "2:00 AM",
      "coupons": 0
    },
    {
      date: "3:00 AM",
      "coupons": 0
    },
    {
      date: "4:00 AM",
      "coupons": 0
    },
    {
      date: "5:00 AM",
      "coupons": 0
    },
    {
      date: "6:00 AM",
      "coupons": 0
    },
    {
      date: "7:00 AM",
      "coupons": 0
    },
    {
      date: "8:00 AM",
      "coupons": 0
    },
    {
      date: "9:00 AM",
      "coupons": 0
    },
    {
      date: "10:00 AM",
      "coupons": 0
    },
    {
      date: "11:00 AM",
      "coupons": 3
    },
    {
      date: "12:00 PM",
      "coupons": 7
    },
    {
      date: "1:00 PM",
      "coupons": 10
    },
    {
      date: "2:00 PM",
      "coupons": 18
    },
    {
      date: "3:00 PM",
      "coupons": 20
    },
    {
      date: "4:00 PM",
      "coupons": 20
    },
    {
      date: "5:00 PM",
      "coupons": 20
    },
    {
      date: "6:00 PM",
      "coupons": 22
    },
    {
      date: "7:00 PM",
      "coupons": 23
    },
    {
      date: "8:00 PM",
      "coupons": 25
    },
    {
      date: "9:00 PM",
      "coupons": 25
    },
    {
      date: "10:00 PM",
      "coupons": 25
    },
    {
      date: "11:00 PM",
      "coupons": 25
    },
    {
      date: "11:59 PM",
      "coupons": 25
    },
  ];
}

function customersData() {
  return [
    {
      date: "12:00 AM",
      "customers": 0
    },
    {
      date: "1:00 AM",
      "customers": 0
    },
    {
      date: "2:00 AM",
      "customers": 0
    },
    {
      date: "3:00 AM",
      "customers": 0
    },
    {
      date: "4:00 AM",
      "customers": 0
    },
    {
      date: "5:00 AM",
      "customers": 0
    },
    {
      date: "6:00 AM",
      "customers": 0
    },
    {
      date: "7:00 AM",
      "customers": 0
    },
    {
      date: "8:00 AM",
      "customers": 0
    },
    {
      date: "9:00 AM",
      "customers": 0
    },
    {
      date: "10:00 AM",
      "customers": 0
    },
    {
      date: "11:00 AM",
      "customers": 3
    },
    {
      date: "12:00 PM",
      "customers": 7
    },
    {
      date: "1:00 PM",
      "customers": 10
    },
    {
      date: "2:00 PM",
      "customers": 18
    },
    {
      date: "3:00 PM",
      "customers": 20
    },
    {
      date: "4:00 PM",
      "customers": 20
    },
    {
      date: "5:00 PM",
      "customers": 20
    },
    {
      date: "6:00 PM",
      "customers": 22
    },
    {
      date: "7:00 PM",
      "customers": 23
    },
    {
      date: "8:00 PM",
      "customers": 25
    },
    {
      date: "9:00 PM",
      "customers": 25
    },
    {
      date: "10:00 PM",
      "customers": 25
    },
    {
      date: "11:00 PM",
      "customers": 25
    },
    {
      date: "11:59 PM",
      "customers": 25
    },
  ];
}

export default { couponData, customersData};