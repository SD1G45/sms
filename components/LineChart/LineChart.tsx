import React from "react";
import {
  LineChart as Chart
} from 'styled-chart'
import { ChartWrapper, StyledHint, StyledPath, StyledTitle, StyledTooltip } from "./styles";
import { LineChartProps } from "./types";

const LineChart: React.FC<LineChartProps> = ({
  title,
  data,
  height,
  flexure
}) => {

  const yAxis = {
    grid: true,
    ticksNum: 0
  };

  const xAxis = {
    grid: true,
    key: "date",
    step: 24,
  };

  const config = {
    "coupons" : {
      label: "Coupons",
      isFilled: false,
      component: (
        <StyledPath 
          strokeColor="#0070f3"
          strokeWidth={2}
        />
      ),
    },
    "customers" : {
      label: "Customers",
      isFilled: false,
      component: (
        <StyledPath 
          strokeColor="#0070f3"
          strokeWidth={2}
        />
      ),
    }
  };

  const tooltip = {
    isVisible: true,
      component: (
        <StyledTooltip backgroundColor="#fff" textColor="#0070f3"/>
      ),
      hints: {
        "coupons": <StyledHint backgroundColor="#0070f3"/>,
        "customers": <StyledHint backgroundColor="#0070f3"/>,
      },
  };

  return (
    <ChartWrapper>
      <StyledTitle>{title}</StyledTitle>
      <Chart
          data={data}
          config={config}
          height={height}
          yAxis={yAxis}
          xAxis={xAxis}
          flexure={flexure}
          tooltip={tooltip}
      />
    </ChartWrapper>
  );
};

export default LineChart;
