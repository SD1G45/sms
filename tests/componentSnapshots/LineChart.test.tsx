/**
 * @jest-environment jsdom
 */

import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import LineChart from "../../components/LineChart";
import sampleData from "../../sampleData/sampleData";

const couponData = sampleData.couponData();

test("LineChart renders correctly", () => {
  const tree = renderer
    .create(
      <div>
        <LineChart
          title="Coupons"
          data={couponData}
          height={300}
          flexure={1}
        />
      </div>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});