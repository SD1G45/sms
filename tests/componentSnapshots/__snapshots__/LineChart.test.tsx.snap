// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChart renders correctly 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 300px;
  width: 100%;
  box-sizing: border-box;
  padding: 8px 8px 48px 8px;
  position: relative;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  position: relative;
}

.c4 {
  display: block;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  width: 100%;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  width: 100%;
}

.c10 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  position: absolute;
  z-index: 1;
  width: calc(100% + 4.166666666666667%);
  margin-left: calc(-2.0833333333333335%);
}

.c16 {
  width: 9px;
  height: 9px;
  border-radius: 50%;
  background: #303030;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  width: 100%;
  bottom: 0;
}

.c14 .c15 {
  opacity: 0;
  position: absolute;
  left: 50%;
  top: 0;
  -webkit-transform: translateX(-50%) translateY(-50%);
  -ms-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
  -webkit-transition: 0.1s all linear;
  transition: 0.1s all linear;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  width: 4%;
}

.c11:hover .c13 .c15 {
  opacity: 1;
}

.c7 {
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  overflow: hidden;
}

.c8 {
  stroke: #757575;
  stroke-width: 2px;
  fill: transparent;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px 0;
  position: absolute;
  white-space: nowrap;
}

.c18 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c18 .c19 {
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 100%;
  margin: 0;
  padding: 0;
  list-style: none;
}

.c6 {
  height: 100%;
  width: 1px;
  background: #e0e0e0;
  pointer-events: none;
  position: absolute;
}

.c9.c9 {
  stroke: #0070f3 !important;
  stroke-width: 2px;
  fill: transparent;
}

.c17.c17 {
  background-color: #0070f3;
}

.c1 {
  text-align: left;
  margin-bottom: -1em;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  gap: 10%;
}

<div>
  <div
    className="c0"
  >
    <text
      className="c1"
    >
      Coupons
    </text>
    <section
      className="c2"
      height="300px"
    >
      <ul
        className="c3"
      />
      <section
        className="c4"
      >
        <section
          className="c5"
        >
          <div
            className="c6"
            style={
              Object {
                "left": "calc(0%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(4.166666666666667%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(8.333333333333334%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(12.5%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(16.666666666666668%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(20.833333333333336%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(25%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(29.166666666666668%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(33.333333333333336%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(37.5%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(41.66666666666667%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(45.833333333333336%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(50%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(54.16666666666667%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(58.333333333333336%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(62.50000000000001%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(66.66666666666667%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(70.83333333333334%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(75%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(79.16666666666667%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(83.33333333333334%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(87.5%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(91.66666666666667%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(95.83333333333334%)",
              }
            }
          />
          <div
            className="c6"
            style={
              Object {
                "left": "calc(100%)",
              }
            }
          />
          <svg
            className="c7"
            style={
              Object {
                "width": "0px",
              }
            }
          >
            <path
              className="c8 c9"
              d="m 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 c 0 0 0 0 0 0 H 100 V 100 H 0"
              strokeWidth={2}
            />
          </svg>
          <div
            className="c10"
            onMouseLeave={[Function]}
          >
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "30%",
                  }
                }
              />
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "30%",
                  }
                }
              />
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "30%",
                  }
                }
              />
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "30%",
                  }
                }
              />
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "30%",
                  }
                }
              />
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "30%",
                  }
                }
              />
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "30%",
                  }
                }
              />
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "30%",
                  }
                }
              />
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "30%",
                  }
                }
              />
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "30%",
                  }
                }
              />
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "30%",
                  }
                }
              />
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "12%",
                  }
                }
              >
                <div
                  className="c13 c14"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <div
                    className="c15 c16 c17"
                  />
                </div>
              </div>
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "28%",
                  }
                }
              >
                <div
                  className="c13 c14"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <div
                    className="c15 c16 c17"
                  />
                </div>
              </div>
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "40%",
                  }
                }
              >
                <div
                  className="c13 c14"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <div
                    className="c15 c16 c17"
                  />
                </div>
              </div>
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "72%",
                  }
                }
              >
                <div
                  className="c13 c14"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <div
                    className="c15 c16 c17"
                  />
                </div>
              </div>
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "80%",
                  }
                }
              >
                <div
                  className="c13 c14"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <div
                    className="c15 c16 c17"
                  />
                </div>
              </div>
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "80%",
                  }
                }
              >
                <div
                  className="c13 c14"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <div
                    className="c15 c16 c17"
                  />
                </div>
              </div>
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "80%",
                  }
                }
              >
                <div
                  className="c13 c14"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <div
                    className="c15 c16 c17"
                  />
                </div>
              </div>
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "88%",
                  }
                }
              >
                <div
                  className="c13 c14"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <div
                    className="c15 c16 c17"
                  />
                </div>
              </div>
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "92%",
                  }
                }
              >
                <div
                  className="c13 c14"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <div
                    className="c15 c16 c17"
                  />
                </div>
              </div>
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "100%",
                  }
                }
              >
                <div
                  className="c13 c14"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <div
                    className="c15 c16 c17"
                  />
                </div>
              </div>
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "100%",
                  }
                }
              >
                <div
                  className="c13 c14"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <div
                    className="c15 c16 c17"
                  />
                </div>
              </div>
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "100%",
                  }
                }
              >
                <div
                  className="c13 c14"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <div
                    className="c15 c16 c17"
                  />
                </div>
              </div>
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "100%",
                  }
                }
              >
                <div
                  className="c13 c14"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <div
                    className="c15 c16 c17"
                  />
                </div>
              </div>
            </div>
            <div
              className="c11"
              onMouseEnter={[Function]}
              width="4%"
            >
              <div
                className="c12"
                onMouseEnter={[Function]}
                style={
                  Object {
                    "height": "100%",
                  }
                }
              >
                <div
                  className="c13 c14"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <div
                    className="c15 c16 c17"
                  />
                </div>
              </div>
            </div>
          </div>
        </section>
        <aside
          className="c18"
        >
          <div
            className="c19 c20"
            style={
              Object {
                "left": "0%",
                "width": "auto",
              }
            }
          >
            12:00 AM
          </div>
          <div
            className="c19 c20"
            style={
              Object {
                "left": "100%",
                "width": "auto",
              }
            }
          >
            11:59 PM
          </div>
        </aside>
      </section>
    </section>
  </div>
</div>
`;
